library/alloc/src/sync/tests.rs

```diff
@@ -140,6 +140,41 @@ fn unwrap_or_drop() {
     assert_eq!(Arc::unwrap_or_drop(x), Some(5));
 }
 
+#[test]
+fn unwrap_or_drop_linked_list() {
```

Comment:
I would either rename this test to show its purpose (e.g. `unwrap_or_drop_no_stackoverflow_on_drop`) or add a comment, else its not very clear what this is trying to do.

Reply:
(The comment can be put later, not necessarily at the top of the function, if you find this way better)

Reply:
I’m not even sure if this test is useful at all. It doesn’t really show much more than the `unwrap_or_drop` test aready does, just applied to a practical example. I think I should just delete it again. I mainly pushed for the docs anyways, and didn’t think about the fact that I also had written this test.

Reply:
Note that code written in docs is compiled and run by default so you can count that as tests.

Reply:
> Note that code written in docs is compiled and run by default so you can count that as tests.

I know, but a single iteration is wayyy unlikely to hit the race condition.

Reply:
> I’m talking about hitting the race condition if the unwrap_or_drop in the test is replaced with try_unwrap + ok

If the call to `try_unwrap + ok` isn't even in the tests, you shouldn't worry about it. Mentioning it clearly (as you have already done) is enough IMO.